test/e2e/secondary_network_test.go

```diff
@@ -0,0 +1,557 @@
+	var annotation_str = ""
+	for _, xPodVN := range listOfVirtualNetworksPerPod {
+		var Input = "{\"name\": \"" + nameOfVirtualNetworkPerPod[pod][xPodVN] + "\" ,\"interface\": \"" + nameOfInterfacePerPod[pod][xPodVN] + "\" , \"type\": \"sriov\"}"
+		if annotation_str == "" {
+			annotation_str = "[" + Input
+		} else {
+			annotation_str = annotation_str + "," + Input
+		}
+	}
+	annotation_str = annotation_str + "]"
+	return annotation_str
+}
+
+func generatePodSpecWithSecondaryNetworkAnnotation() error {
+	// Generating yaml files for all the pods
+	for xPod := 0; xPod < totalNumberOfPods; xPod++ {
+		annotationString := formAnnotationStringOfPod(xPod)
+		// Formation of Pod Spec, Metadata, kind and versionme
+		podSpecObj := PodSpec{APIVersion: "v1",
+			Kind:     "Pod",
+			Metadata: Metadata{Annotations: Annotations{K8SV1CniCncfIoNetworks: annotationString}, Labels: Labels{App: "testsecpod"}, Name: nameOfPods[xPod]},
+			Spec: Spec{Containers: [1]Containers{{Image: "busybox", ImagePullPolicy: "IfNotPresent", Command: []string{"sleep"}, Args: []string{"infinity"}, Name: "busyboxpod",
+				Resources: Resources{Requests: Requests{IntelComIntelSriovNetdevice: "3"}, Limits: Limits{IntelComIntelSriovNetdevice: "3"}}}}, RestartPolicy: "OnFailure"}}
+
+		data, err := yaml.Marshal(&podSpecObj)
+		if err != nil {
+			log.Fatal(err)
+		}
+		// Writing into the pod yaml file
+		error2 := ioutil.WriteFile(nameOfYamlPerPod[xPod], data, 0)
+		if error2 != nil {
+			fmt.Errorf("Error in writing to yaml file! Please Check.\n")
+			log.Fatal(error2)
+		}
+	}
+	return nil
+}
+func (data *TestData) createSecondaryNetworkPods() error {
+	for _, YamlName := range nameOfYamlPerPod {
+		respCode, _, _, err := data.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl apply -f %s -n kube-system", YamlName))
+		if err != nil || respCode != 0 {
+			return fmt.Errorf("Error when creating %s secondary test pods - rc: %v, err: %v", YamlName, respCode, err)
+		}
+	}
+	return nil
+}
+
+func (data *TestData) deleteSecondaryNetworkPods() error {
+	for _, podName := range nameOfPods {
+		respCode, _, _, err := data.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl delete pod -f %s -n kube-system", podName))
+		if err != nil || respCode != 0 {
+			return fmt.Errorf("Error when deleting %s secondary test pods - rc: %v, err: %v", podName, respCode, err)
+		}
+	}
+	return nil
+}
+
+func (data *TestData) checkSecondaryInterfaces() error {
+	for _, PodName := range nameOfPods {
+		respCode, _, _, err := data.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl exec -it %s -n kube-system -- ip addr show", PodName))
+		if err != nil || respCode != 0 {
+			return fmt.Errorf("Error when creating interfaces on %s remote cluster - rc: %v, err: %v", PodName, respCode, err)
+		}
+	}
+	return nil
+}
+
+func (data *TestData) generateAntreaConfig() error {
+	var err error
+	genManifest := "./../../hack/generate-manifest.sh"
```

Comment:
I mentioned it in another comment in response to @jianjuns' question, but I think all of this should happen outside of the test case.

IMO, the flow should look like this:
* `./deploy-antrea.sh` (deploy Antrea and dependencies to a cluster for secondary network testing)
* `go test ./e2e/secondary_network -config <>`